src/features/game/lib/mapMovement.ts

```diff
@@ -0,0 +1,71 @@
+type MovementIntervals = {
+  moveUp?: any,
+  moveDown?: any,
+  moveLeft?: any,
+  moveRight?: any
+}
+
+
+const sensitivity = 3
+const movementIntervals: MovementIntervals = {}
+const initialCoordinates = [1024,1214]
+// TODO dynamic initial coordinates based on tile dimensions
+
+const setup = (container: any) => {
+  // adding setintervals for smooth movement for each relevant keypress and removing them on keyup 
+  window.addEventListener('keydown', (event) => {
```

Comment:
We've got 2 window listeners here that don't get cleaned up.

When a user changes their Metamask account we will take them back to the login page. Once they log back into the game 2 more window listeners would be added. This will cause some strange behaviour with too many listeners.

I would suggest adding some code that cleans up these listeners when the component (`Session.tsx`) is unmounted.

You can achieve this in the components use effect by providing a disposer function.

```
useEffect(() => {
	// do some code

       return () => {
          // clean up listeners
      }
})
```